Mage/src/main/java/mage/abilities/effects/common/DrawCardSourceControllerEffect.java

```diff
@@ -57,16 +70,34 @@ public boolean apply(Game game, Ability source) {
         return false;
     }
 
-    private void createStaticText(String whoDrawCard) {
-        StringBuilder sb = new StringBuilder(whoDrawCard);
-        sb.append(whoDrawCard.isEmpty() ? "draw " : " draw ");
-        String value = amount.toString();
-        sb.append(CardUtil.numberToText(value, "a"));
-        sb.append(value.equals("1") ? " card" : " cards");
-        String message = amount.getMessage();
-        if (!message.isEmpty()) {
-            sb.append(value.equals("X") ? ", where X is " : " for each ");
-            sb.append(message);
+    private void createStaticText(boolean youDraw, DrawCardsPhrasing drawCardsPhrasing) {
+        StringBuilder sb = new StringBuilder(youDraw ? "you " : "");
+        sb.append("draw");
+        String value = " a";
+        if (amount instanceof StaticValue) {
+            value = " " + CardUtil.numberToText(((StaticValue)amount).getValue(), "a");
+        } else if (drawCardsPhrasing == DrawCardsPhrasing.X_WHERE) {
+            value = " X";
+        } else if (drawCardsPhrasing == DrawCardsPhrasing.EQUAL_TO) {
+            value = "";
+        } else if (amount instanceof MultipliedValue) {
```

Comment:
generally not a fan of these instanceof checks, but open to admitting that it's the least bad approach

Reply:
This approach is probably going to lead to making a `DecoratedValue` abstract class (or interface?) for instance checking like this, since all the decorated values will need their qualification put into the text in this manner.

Another possible approach is to make decorator classes and `StaticValue` inherit a common class or interface (maybe still named `DecoratedValue`, though having `StaticValue` inherit that seems wrong to me) that has a `getValue()` function (or something to descriptively replace `toString()`) that goes in place here in the first brackets:
Draw <a/two/one plus X/> cards <for each .../where X is ...>

Thoughts?

Reply:
I think it should be possible to have the method in DynamicValue itself

Reply:
On second thought, i'm not sure how to avoid the `instanceof` checks. Technically, some unique behavior could be implemented that identifies instances of `MultipliedValue` and `StaticValue` such that they can be handled in places like this, but i'm realizing two things that make this kinda pointless:
1. `StaticValue`, `MultipliedValue`, and the other `DynamicValue` implementations must all be handled differently (3 broad cases)
2. I can't think of any other cases than these three. the other decorator classes haven't necessitated property handling like 'MultipliedValue' has. `AdditiveDynamicValue` and `IntPlusDynamicValue` only need to modify their `getMessage` calls, and `LimitedDynamicValue` and `LimitedDynamicValue` are used in one place each, so trying to generalize text for them doesn't seem worth it.

So trying to avoid using `instanceof` will still land us handling these classes specially, at which point why not use `instanceof`